### ISSUE TITLE
MAG Child Resource Retrieval Issue

### ISSUE THREAD

**Issue by robertdias** [microsoft/AzureStorageExplorer#405]
**Storage Explorer Version**: 1.3.0
**Platform/OS Version**: Win10
**Architecture**: 
**Regression From**: 1.2.0

#### Steps to Reproduce: ####
1. Access Azure US Government account.
2. Open any Blob Container or File Share

#### Expected Experience: ####
Hosted resources are displayed.

#### Actual Experience: ####
Error is thrown: Unable to retrieve child resources

Details: The value for one of the HTTP headers is not in the correct format.
![imgo](https://user-images.githubusercontent.com/34075421/42596848-0c02d636-8525-11e8-8194-64ceaf6e9030.jpg)

**Comment by MRayermannMSFT** [comment]
Hi @robertdias ,are you accessing the account via sign, SAS, or name and key?

**Comment by MRayermannMSFT** [comment]
Did you select US Government as you endpoints domain?
![image](https://user-images.githubusercontent.com/21112697/42606413-46bc93d6-8532-11e8-874c-72a9f9854944.png)

**Comment by robertdias** [comment]
I’m sorry. I meant I use a username and password to access

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Comment by MRayermannMSFT** [comment]
Hmmmm unfortunately I am not seeing this issue. Have you tried removing and reading the account?

**Comment by robertdias** [comment]
Yes. I tried that before reverting to v1.2.0. 

Other members on our workstream have experienced the same issue.

**Comment by MRayermannMSFT** [comment]
Do y'all use Visual Studio? If so, is Cloud Explorer able to enumerate your Storage resources?

**Comment by iBobb** [comment]
I am signed into the explorer using my company's domain microsoft account we use for Azure. After update 1.3.0 my quick access to a File Shares location got broken but I cannot even access it from the Storage accounts when I find it there in the list. I get the same error message. Today I updated to 1.3.1 and the issue persists. Of course in the Azure portal I can access this same location without a problem.

**Comment by MRayermannMSFT** [comment]
@iBobb , are you signing into Azure or Azure US Government?

**Comment by madmox** [comment]
Same issue here. We can't access some of our file shares from azure storage explorer as of version 1.3.1 (but previous installed version was 1.2.0).

Not all our file storage accounts are impacted. The error is the same as the one in the screenshot above.

**Comment by MRayermannMSFT** [comment]
If someone could please do the following, it'd be very helpful:
1. Open the dev tools (F12)
2. Use ctrl+p to open `NodeGroupViewModel.js`
3. Set a breakpoint on line 82 by clicking on "82" (the line should contain "Unable to retrieve child resources")
4. Repro the error
5. When the breakpoint is hit, hover over `err` to see expanded error info, and comment here with what you see. What we're looking for is information as to what header is invalid.

**Comment by madmox** [comment]
Done. I truncated the staktrace but kept other useful infos:

```
code:"InvalidHeaderValue"
headername:"x-ms-version"
headervalue:"2018-03-28"
name:"StorageError"
requestId:"c42dbc80-d01a-00d1-1bd5-1e7d33000000"
statusCode:400
message:"The value for one of the HTTP headers is not in the correct format. RequestId:c42dbc80-d01a-00d1-1bd5-1e7d33000000 Time:2018-07-18T20:25:40.3752558Z"
stack:"StorageError: The value for one of the HTTP headers is not in the correct format. [...]"
```

**Comment by MRayermannMSFT** [comment]
Thank you @madmox ! Wish I had just asked for that from the get go. I'm going to reach out to some people and try to find out why that is an invalid version ASAP. Thanks again. 😄

**Comment by MRayermannMSFT** [comment]
Hey @madmox , @iBobb , @robertdias , thank you for your patience. Turns out Azure US Govt was doing a delayed/gradual rollout of the 2018-03-28 Storage API version. We were unfortunately not made aware of that decision. This is why we were unable to repro this issue/didn't run into during testing. Our test resources were located where the update had already happened, while your resources' locations had not received the update. I've been told that the update will finish in the next 48 hours, but if you would like to be unblocked now, or if the update does actually not finish in the next 48 hours you can do the following:

1. Open your favorite editor as admin (if you use VS Code, you can just open normally)
2. Open "C:\Program Files (x86)\Microsoft Azure Storage Explorer\resources\app\out\app\renderer\StorageApiSettingManager.js" (you path may differ depending on install location
3. On line 31, change `var storageApiVersion = this.getStorageApiSetting();` to `var storageApiVersion = "2017-11-09";`

Apologies for this issue, we've let the team responsible for the rollout know that in similar future situations we need to be informed so we can pass that information on to you/make sure we give y'all a good experience.

Finally, I suggest only doing theworkaround while the update is still going on, as you will get the best stability/behavior when using the most recent API version.

**Comment by robertdias** [comment]
Thank you!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by madmox** [comment]
Just to clarify : I encountered the issue on Azure West Europe storage accounts, nothing related to Azure US Government. And some of our storage accounts in the same region were fine.

**Comment by MRayermannMSFT** [comment]
@madmox , they must have been doing a gradual rollout as well. I'm not seeing the issue with our test subscriptions, but if you still are feel free to do the workaround, and if you'd like, I can try to find out when the rollout is planned to be complete.

**Comment by MRayermannMSFT** [comment]
I believe this all should be good now. If anyone is still having problems please open a new issue.
